### ISSUE TITLE
[Question or Issue ?] Authentication provider errror when using a httpProvider without authentication provider

### ISSUE THREAD

**Issue by Mahoneko** [microsoftgraph/msgraph-sdk-dotnet#362]
Please provide the following (and please check them off the list with [x]) before submitting this issue:
- [ ] Expected behavior. Please provide **links to the specific [Microsoft Graph documentation](https://developer.microsoft.com/en-us/graph/docs/concepts/overview)** you used to determine the expected behavior. 
- [x] Actual behavior. Provide error codes, stack information, and a [Fiddler](http://www.telerik.com/fiddler) capture of the request and response (please remove personally identifiable information before posting).
- [x] Steps to reproduce the behavior. Include your code, IDE versions, client library versions, and any other information that might be helpful to understand your scenario.

### Description

I'm using SDK .Net in an ASP.Net Core 2.2 project.

Requests have to pass through a Http proxy, so I provided a HttpProvider with proxy configuration to the GraphServiceClient to do that.

The below code was working in 1.12.0 but not working anymore in 1.13.0.

I don't know if I'm doing something wrong.

### Expected behavior

I don't know if it's an issue or a change of the way to use GraphServiceClient with DelegateAuthenticateHandler and HttpProvider

If now provided HttpProvider must have the AuthenticationProvider, the following constructor become useless :
> GraphServiceClient(DelegateAuthenticateHandler, HttpProvider)

Constructor with HttpProvider does not need AuthenticationProvider
> GraphServiceClient(HttpProvider)

### Actual behavior

```
Microsoft.Graph.ServiceException: Code: invalidRequest
Message: Authentication provider is required before sending a request.

   at Microsoft.Graph.HttpProvider.SendRequestAsync(HttpRequestMessage request, HttpCompletionOption completionOption, CancellationToken cancellationToken)
   at Microsoft.Graph.HttpProvider.SendAsync(HttpRequestMessage request, HttpCompletionOption completionOption, CancellationToken cancellationToken)
   at Microsoft.Graph.BaseRequest.SendRequestAsync(Object serializableObject, CancellationToken cancellationToken, HttpCompletionOption completionOption)
   at Microsoft.Graph.BaseRequest.SendAsync[T](Object serializableObject, CancellationToken cancellationToken, HttpCompletionOption completionOption)
   at Microsoft.Graph.GraphServiceUsersCollectionRequest.GetAsync(CancellationToken cancellationToken)
   at ADIS.WebApi.Common.Helpers.MicrosoftGraphManagerUsingSDK.GetUser(String upn, Expression`1 selectExpression)
```

### Steps to reproduce the behavior

ASP.Net Core 2.2 WebApi
Microsoft.Graph 1.13.0

Proxy is not mandatory to reproduce error, just using a httpProvider without authentication handler

##### Working in 1.12.0 but not in 1.13.0

```C#

HttpProvider httpProvider = new HttpProvider();

// Init MS Graph client
_graphClient = new GraphServiceClient(
    new DelegateAuthenticationProvider(requestMessage =>
    {
        requestMessage.Headers.Authorization = new AuthenticationHeaderValue("bearer", _accessToken);
        return Task.FromResult(0);
    }), httpProvider);

```

Then make any request

##### Working in 1.12.0 and in 1.13.0

I'm able to fix it by moving the authentication handler to HttpProvider:

```C#

HttpProvider httpProvider = new HttpProvider(
    new DelegateAuthenticationProvider(requestMessage =>
    {
        requestMessage.Headers.Authorization = new AuthenticationHeaderValue("bearer", _accessToken);
        return Task.FromResult(0);
    })
);

// Init MS Graph client
_graphClient = new GraphServiceClient(null, httpProvider);

```

Then make any request

Edit : seems related to this commit : https://github.com/microsoftgraph/msgraph-sdk-dotnet/commit/e97039d5920fe726220eb7ba0a1cfeeaa6ceeda8#diff-ae7fba9d494bd6e152a9340223a114c7

**Comment by pwombwa** [comment]
@Mahoneko Thank you for bringing this up. We have delisted this version and a fix will be provided in the next update. Please use version 1.12.0 for now.

**Comment by columbusIT-mgr** [comment]
Any news on the fix? 
The 1.12.0 version of API does not work with NewtonSoft Json 12.0.1
Any workaround to make the 1.13.0 version properly functioning on this issue?

**Comment by pwombwa** [comment]
@columbusIT-mgr , @Mahoneko We have released 1.14.0-preview which has a fix for this. Let us know if it addresses your particular scenarios.
